### PR TITLE
fix(core): Reorder `field_validator` and `classmethod` to fit Pydantic V2.

### PR DESCRIPTION
**Pull request by laipz8200**
# Description

1. Move the `@classmethod` below. [See reference](https://docs.pydantic.dev/2.7/concepts/validators/#field-validators)
2. Add `preview = true` to the ruff lint configuration.
3. Fix the Pydantic validation error in `HttpRequestNodeData`

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Improvement, including but not limited to code refactoring, performance optimization, and UI/UX improvement

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] By running test srcipts.

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods
- [ ] `optional` I have made corresponding changes to the documentation 
- [ ] `optional` I have added tests that prove my fix is effective or that my feature works
- [ ] `optional` New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by bowenliang123** [comment]
-1 for this. 
Its order is placed by design.

If `@classmethod` is placed after `@field_validator`,  the warning message shows on IDEA.
image

**Comment by bowenliang123** [comment]
As for the second point `Add preview = true to the ruff lint configuration.`, feel free to remove all the `--preview` options in hooks, CI style jobs and `./dev` scripts.

As for the third point, thanks for fixing it, if any concrete evidence for comparing the changes.

**Comment by laipz8200** [comment]
> If `@classmethod` is placed after `@field_validator`, the warning message shows on IDEA. image
> 
> But it's reasonable if we have to follow the official recommendation.

image

Based on my experience, JetBrains IDEs aren't that strong in type checking. You might want to give Pyright or Pylance a try for better results.

**Comment by laipz8200** [comment]
I've removed the `--preview` flag from the CI and reformat scripts.

The third point is about fixing the changes we didn't complete when migrating to v2.x. The new parameter structure can be found [here](https://docs.pydantic.dev/2.7/api/pydantic_core_schema/#pydantic_core.core_schema.ValidationInfo).

File: .github/workflows/style.yml [hunk 1]
```diff
@@ -31,15 +31,15 @@ jobs:
         uses: actions/setup-python@v5
         if: steps.changed-files.outputs.any_changed == 'true'
         with:
-          python-version: '3.10'
+          python-version: "3.10"
```
Comment (bowenliang123) [inline .github/workflows/style.yml]: Unnecessary changes.

File: dev/reformat [hunk 2]
```diff
@@ -3,18 +3,18 @@
 set -x
 
 # python style checks rely on `ruff` in path
-if ! command -v ruff &> /dev/null; then
-    echo "Installing Ruff ..."
-    pip install ruff
+if ! command -v ruff &>/dev/null; then
+	echo "Installing Ruff ..."
+	pip install ruff
```
Comment (bowenliang123) [inline dev/reformat]: Unnecessary and unrelated changes, which introduced incorrect indent. Please leave it untouched.

File: dev/reformat [hunk 3]
```diff
@@ -3,18 +3,18 @@
...
-ruff check --fix --preview ./api
+ruff check --fix ./api
 
 # env files linting relies on `dotenv-linter` in path
-if ! command -v dotenv-linter &> /dev/null; then
-    echo "Installing dotenv-linter ..."
-    pip install dotenv-linter
+if ! command -v dotenv-linter &>/dev/null; then
+	echo "Installing dotenv-linter ..."
+	pip install dotenv-linter
```
Comment (bowenliang123) [inline dev/reformat]: ditto.

**Review by bowenliang123 — APPROVED** [review]
LGTM overall. Please follow the minor comments.

**Review by ghost — APPROVED** [review]

**Review by ghost — APPROVED** [review]

File: .github/workflows/style.yml (hunk 1 above)
Comment (laipz8200) [reply .github/workflows/style.yml]: Oops! My linter caused that. I've changed it back.

File: dev/reformat (hunk 2 above)
Comment (laipz8200) [reply dev/reformat]: Updated.

File: dev/reformat (hunk 3 above)
Comment (laipz8200) [reply dev/reformat]: Updated.

**Review by takatost — APPROVED** [review]
LGTM
